sdk/storage/azure-storage-blob/azure/storage/blob/_download.py

```diff
@@ -718,17 +721,20 @@ def read(self, size: int = -1, *, chars: Optional[int] = None) -> T:
                 self._complete_read()
 
             else:
-                while (chunk := next(chunks_iter, None)) is not None and remaining > 0:
+                while (chunk := next(chunks_iter, None)) is not None and remaining > 0:  # type: ignore [arg-type]
                     chunk_data, content_length = downloader.yield_chunk(chunk)
                     self._download_offset += len(chunk_data)
                     self._raw_download_offset += content_length
-                    self._current_content = self._decoder.decode(
-                        chunk_data, final=self._download_complete) if self._text_mode else chunk_data
+                    if self._text_mode and self._decoder is not None:
+                        self._current_content = self._decoder.decode(
+                        chunk_data, final=self._download_complete)
+                    else:
+                        self._current_content = chunk_data
 
                     if remaining < len(self._current_content):
-                        read = output_stream.write(self._current_content[:remaining])
+                        read = output_stream.write(self._current_content[:remaining])  # type: ignore [arg-type]
                     else:
-                        read = output_stream.write(self._current_content)
+                        read = output_stream.write(self._current_content)  # type: ignore [arg-type]
```

Comment:
azure\storage\blob\_download.py:742: error: Argument 1 to "write" of "BufferedIOBase" has incompatible type "Union[str, bytes]"; expected "Buffer"  [arg-type]
azure\storage\blob\_download.py:742: error: Argument 1 to "write" of "TextIOBase" has incompatible type "Union[str, bytes]"; expected "str"  [arg-type]